Lib/asyncio/tasks.py

```diff
@@ -42,9 +42,19 @@ def all_tasks(loop=None):
     """Return a set of all tasks for the loop."""
     if loop is None:
         loop = events.get_running_loop()
-    # NB: set(_all_tasks) is required to protect
+    # NB: list(_all_tasks) is required to protect
     # from https://bugs.python.org/issue34970 bug
-    return {t for t in list(_all_tasks)
+    # NB: Have to repeat on RuntimeError, other thread
```

Comment:
There's no point in prefixing all comments with "NB"; it's just a regular comment.  I'd suggest to write something along the following (as it took me sometime to understand why this change is necessary at all):

"Looping over a WeakSet (_all_tasks) isn't safe as it can be updated from another thread while we do so. Therefore we cast it to list prior to filtering. The list cast itself requires iteration, so we repeat it several times ignoring RuntimeErrors (which are not very likely to occur).  See issues 34970 and 36607 for details."